src/DomUtils/elementPositionInRow.ts

```diff
@@ -143,28 +143,36 @@ export function getElementInPrevRow(
 
 export function firstVisibleElementInContainer(
   parent: NullableElement,
-  elements: HTMLElement[]
+  elements: HTMLElement[],
+  maxVisibilityDiffThreshold = 0
 ): NullableElement {
   if (!parent || !elements.length) {
     return null;
   }
 
   const parentTop = parent.getBoundingClientRect().top;
   const parentBottom = parent.getBoundingClientRect().bottom;
-
-  const parentTopWithLabel = parentTop + getLabelHight(parent);
+  const parentTopWithLabel = parentTop + getLabelHeight(parent);
 
   const visibleElements = elements.find(element => {
     const elementTop = element.getBoundingClientRect().top;
     const elementBottom = element.getBoundingClientRect().bottom;
+    const maxVisibilityDiffPixels =
+      element.clientHeight * maxVisibilityDiffThreshold;
 
-    if (elementTop < parentTopWithLabel) {
+    const elementTopWithAllowedDiff = elementTop + maxVisibilityDiffPixels;
+    const elementBottomWithAllowedDiff =
+      elementBottom - maxVisibilityDiffPixels;
+
+    if (elementTopWithAllowedDiff < parentTopWithLabel) {
       return false;
     }
 
     return (
-      (elementTop >= parentTop && elementTop <= parentBottom) ||
-      (elementBottom >= parentTop && elementBottom <= parentBottom)
+      (elementTopWithAllowedDiff >= parentTop &&
+        elementTopWithAllowedDiff <= parentBottom) ||
+      (elementBottomWithAllowedDiff >= parentTop &&
+        elementBottomWithAllowedDiff <= parentBottom)
     );
```

Comment:
after fixing `getLabelHeight` logic, the remaining issue was that this check didn't pass because of the 1px difference caused by the `scrollTo` function, i wasn't sure why this 1px offset was added `$eprBody.scrollTop = top + 1;` so instead of removing it (fixes the issue as well) i thought it's a good idea to introduce an optional threshold for visibility calculations

export function scrollTo(root: NullableElement, top: number = 0) {
  const $eprBody = queryScrollBody(root);

  if (!$eprBody) {
    return;
  }

  requestAnimationFrame(() => {
    $eprBody.scrollTop = top + 1;
  });
}

Reply:
That's a good point. One thing, I just checked, and the reason that I needed that 1px is no longer relevant. Back then it messed up my active category calculation. Seems like it works well without it. I'll remove it soon enough :)